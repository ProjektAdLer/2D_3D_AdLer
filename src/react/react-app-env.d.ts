/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
