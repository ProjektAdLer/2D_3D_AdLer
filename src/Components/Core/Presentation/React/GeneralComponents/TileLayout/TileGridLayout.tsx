import Tile from "./Tile";

type TileLayoutProps = {
  tileContents: {
    id: number;
    title?: string;
    image: string;
  }[];
  columns: number;
  mobileColumns?: number;
  onTileClick: (id: number) => void;
};

export default function TileGridLayout(props: TileLayoutProps) {
  return (
    <div
      className={`grid portrait:grid-cols-${props.mobileColumns ?? props.columns} grid-cols-${props.columns} gap-5 pt-4 bg-whitetrans rounded-lg p-4 overflow-y-auto`}
    >
      {props.tileContents.map((tile) => (
        <Tile
          key={tile.id}
          title={tile.title}
          image={tile.image}
          onClick={() => props.onTileClick(tile.id)}
        />
      ))}
    </div>
  );
}
