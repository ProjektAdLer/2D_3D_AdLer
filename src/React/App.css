body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
