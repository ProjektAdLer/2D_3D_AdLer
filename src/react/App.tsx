import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
