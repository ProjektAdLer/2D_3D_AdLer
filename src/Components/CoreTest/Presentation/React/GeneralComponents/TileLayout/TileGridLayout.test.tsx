import { render } from "@testing-library/react";
import React from "react";
import TileGridLayout from "../../../../../Core/Presentation/React/GeneralComponents/TileLayout/TileGridLayout";

describe("TileGridLayout", () => {
  test("should render", () => {
    const contents = [
      {
        id: 1,
        title: "title",
        image: "image",
      },
    ];

    const renderResult = render(
      <TileGridLayout
        tileContents={contents}
        columns={2}
        onTileClick={() => {}}
      />,
    );
    expect(renderResult.container).toMatchSnapshot();
  });

  test("calls onTileClick with id when tile is clicked", () => {
    const contents = [
      {
        id: 1,
        title: "title",
        image: "image",
      },
      {
        id: 2,
        title: "title",
        image: "image",
      },
    ];

    const onTileClick = jest.fn();
    const renderResult = render(
      <TileGridLayout
        tileContents={contents}
        columns={2}
        onTileClick={onTileClick}
      />,
    );

    renderResult.getAllByRole("button")[0].click();

    expect(onTileClick).toHaveBeenCalledTimes(1);
    expect(onTileClick).toHaveBeenCalledWith(1);
  });
});
