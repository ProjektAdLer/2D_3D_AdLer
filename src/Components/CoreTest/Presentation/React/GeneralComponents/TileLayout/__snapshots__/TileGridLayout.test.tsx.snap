// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileGridLayout should render 1`] = `
<div>
  <div
    class="grid portrait:grid-cols-2 grid-cols-2 gap-5 pt-4 rounded-lg p-4"
  >
    <div
      class=""
    >
      <button
        class="flex-col px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue landscape:hover:bg-adleryellow active:bg-adleryellow"
      >
        <div
          class="flex flex-col"
        >
          <img
            alt="titleThumbnail"
            src="image"
          />
          <p>
            title
          </p>
        </div>
      </button>
    </div>
  </div>
</div>
`;
